extension/src/extension.ts

```diff
@@ -545,25 +508,33 @@ export async function activate(ctx: sourcegraph.ExtensionContext): Promise<void>
                 if (canGenerateTraceUrl(span)) {
                     logger.log('Hover trace', span.generateTraceURL())
                 }
-                const textDocumentUri = new URL(textDocument.uri)
-                const serverRootUri = resolveServerRootUri(textDocumentUri, serverSgEndpoint)
-                const serverTextDocumentUri = toServerTextDocumentUri(textDocumentUri, serverSgEndpoint)
-
-                const connection = await getOrCreateConnection(serverRootUri, { span, token })
-                // Poll server to get updated results when e.g. dependency installation finished
-                while (true) {
-                    const hoverResult = await sendTracedRequest(
-                        connection,
-                        HoverRequest.type,
-                        {
-                            textDocument: { uri: serverTextDocumentUri.href },
-                            position,
-                        },
-                        { span, tracer, token }
-                    )
-                    rewriteUris(hoverResult, toSourcegraphTextDocumentUri)
-                    yield convertHover(hoverResult)
-                    await delayPromise(HOVER_DEF_POLL_INTERVAL)
+
+                if (await isLSIFAvailable(textDocument)) {
+                    const lsifResult = await lsif.hover(textDocument, position)
+                    yield lsifResult && lsifResult.value
+                } else if (!config.value['typescript.serverUrl']) {
+                    yield await basicCodeIntel.hover(textDocument, position)
```

Comment:
(not blocking) it would be great if basic code intel and LSIF functions took an OpenTracing Span to propagate tracing to the backend. That would allow us to measure performance of codeintel implementations end-to-end.

Reply:
:+1: I'll add OpenTracing to the perf testing doc